### DISCUSSION TITLE
Anyone know of any Web App Firewall rules or something else blocking httpx user-agent?

### DISCUSSION THREAD

**Discussion by Jaharmi** [encode/httpx#2684] [Potential Issue]
I recently had a script stop working. Investigation revealed that the destination server returned `403 Forbidden` if the `user-agent` was set to HTTPX. The problem happened somewhere between last week and earlier this week (around Tuesday, 2023-04-25).

I was able to verify this by changing the user-agent in my script, which worked. I also tried the same server API call with Curl set to mimic the HTTPX user-agent, which gave the same HTTP error response.

```
User-Agent: python-https/0.24.0
```

It struck me that there may be some vendor/cloud/community WAF rules or similar that are blocking — improperly, I'd say — on the HTTPX user-agent. If so, I figured others may have run into this issue.

**Comment by lovelydinosaur** [comment]
> If so, I figured others may have run into this issue.

Yeah can confirm that I've seen the `httpx` User-Agent blocked once before.
I don't recall which URL it was for, but changing the User-Agent header resolved it.

Bit frustrating really.
